gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,227 @@
+    return fluid.promise.fireTransformEvent(onCaptureSettingsForCurrentDevice,
+        {}, options);
+};
+
+/**
+ * This function invokers a settings handler to fetch settings, but wraps the processing and reports
+ * back any failures in an `isError: true` json block.
+ *
+ * @param {Function} invokeSettingsHandlerGet - The get function from the appropriate settings handler.
+ * @param {String} solutionID - The dotted solution id, ex `com.freedomscientific.jaws`
+ * @param {Object} handlerSpec - TODO, exactly how much of the solutions entry block is this?
+ * @return {Object} The returned settings, or an error block with debugging information.
+ */
+gpii.flowManager.capture.safeHandlerGet = function (invokeSettingsHandlerGet, solutionID, handlerSpec) {
+    var promiseTogo = fluid.promise();
+    try {
+        invokeSettingsHandlerGet(solutionID, handlerSpec).then(function (data) {
+            promiseTogo.resolve(data);
+        }, function (err) {
+            promiseTogo.resolve({
+                solutionId: solutionID,
+                handlerSpec: handlerSpec,
+                isError: true,
+                message: "Capture Workflow Error: Settings Handler Promise Rejection",
+                err: err
+            });
+        });
+    }
+    catch (err) {
+        promiseTogo.resolve({
+            solutionId: solutionID,
+            handlerSpec: handlerSpec,
+            isError: true,
+            message: "Capture Workflow Error: Unexpected Error in operating settings handler.",
+            err: err
+        });
+    }
+    return promiseTogo;
+};
+
+/**
+ * Runs through all the solutions currently available on the system, pulls the current
+ * setting for each supportedSetting and returns them in an object. Primary use case
+ * is for backing Capture tools that would allow a user to set up their GPII profile
+ * starting with the current settings for their applications on the local machine.
+ *
+ * @param {gpii.lifecycleManager} lifecycleManager - Instance of the local machine lifecycle manager.
+ * @param {Object} solutions - Solutions registry entries for solutions available on the current machine.
+ * @param {Object} options - Extra options for processing.
+ * @param {Array} options.solutionsList - If provided, only solutions in this list of `solutionsID`s will
+ * be captured. Example:
+ *
+ * '''json
+ * ["com.microsoft.windows.cursors", "com.freedomscientific.jaws"]
+ * '''
+ * @return {fluid.promise} Returns a promise resolving with the entire system settings capture.
+ */
+gpii.flowManager.captureSystemSettings = function (lifecycleManager, solutions, options) {
+    var solutionsToFetch = fluid.copy(solutions);
+    if (options.solutionsList) {
+        fluid.remove_if(solutionsToFetch, function (solution, solutionID) {
+            return !options.solutionsList.includes(solutionID);
+        });
+    }
+
+    return lifecycleManager.processRead(solutionsToFetch);
```

Comment:
What do you think to pass "{lifecycleManager}.read" as the first argument into this function to narrow down the scope of this argument? Thanks.

Reply:
Sure why not :)  Updated to take in this read function.